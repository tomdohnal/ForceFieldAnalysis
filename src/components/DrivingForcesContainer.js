// @flow
import React from 'react';

import ForcesContainer from '../containers/ForcesContainer';

const DrivingForcesContainer = () => (
  <ForcesContainer submitButtonText="Create Driving Force" driving />
);

export default DrivingForcesContainer;
