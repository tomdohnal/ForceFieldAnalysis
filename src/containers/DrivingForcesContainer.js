// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Input, Button } from 'semantic-ui-react';

import DrivingForceArrow from '../components/DrivingForceArrow';
import { addDrivingForce, type Forces } from '../ducks/forces';

type State = {
  newDrivingForceName: string,
  newDrivingForceError: boolean,
  newDrivingForceErrorText: string,
};

type Props = {
  forces: Forces,
  addDrivingForce: (name: string, strength: number) => void,
};

class DrivingForcesContainer extends Component<Props, State> {
  state = {
    newDrivingForceName: '',
    newDrivingForceError: false,
    newDrivingForceErrorText: '',
  };

  onNewDrivingForceInputChange = (e) => {
    this.setState({
      newDrivingForceName: e.target.value,
      newDrivingForceError: false,
      newDrivingForceErrorText: '',
    });
  };

  onNewDrivingForceButtonClick = () => {
    if (!this.state.newDrivingForceName) {
      this.setState({
        newDrivingForceError: true,
        newDrivingForceErrorText: 'Enter the description for the force.',
      });
    } else {
      this.setState({
        newDrivingForceName: '',
      });

      this.props.addDrivingForce(this.state.newDrivingForceName, 2);
    }
  };

  renderForces() {
    return this.props.forces.map(force => (
      <DrivingForceArrow strength={force.strength} key={force.id} />));
  }

  render() {
    return (
      <div>
        {this.renderForces()}
        <Input
          onChange={this.onNewDrivingForceInputChange}
          value={this.state.newDrivingForceName}
          error={this.state.newDrivingForceError}
        />
        <Button onClick={this.onNewDrivingForceButtonClick}>New driving force</Button>
        <span>{this.state.newDrivingForceErrorText}</span>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  forces: state.forces.filter(force => force.driving === true),
});

export default connect(mapStateToProps, { addDrivingForce })(DrivingForcesContainer);
