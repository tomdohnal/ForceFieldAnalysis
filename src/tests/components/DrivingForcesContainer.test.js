import React from 'react';
import { shallow } from 'enzyme';

import DrivingForcesContainer from '../../components/DrivingForcesContainer';

describe('DrivingForcesContainer', () => {
  it('renders correctly', () => {
    const tree = shallow(<DrivingForcesContainer text="Create Driving Force" />);

    expect(tree).toMatchSnapshot();
  });
});
