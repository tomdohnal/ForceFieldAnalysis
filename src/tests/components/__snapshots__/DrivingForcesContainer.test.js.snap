// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrivingForcesContainer renders correctly 1`] = `
<Connect(ForcesContainer)
  driving={true}
  submitButtonText="Create Driving Force"
/>
`;
